src/Oracles/ModuleFiles.hs

```diff
@@ -136,7 +141,16 @@ moduleFilesOracle = void $ do
                 let cmp f = compare (dropExtension f)
                     found = intersectOrd cmp files mFiles
                 return (map (fullDir -/-) found, mDir)
-        let pairs = sort [ (encodeModule d f, f) | (fs, d) <- result, f <- fs ]
+
+        mainpairs <- case mainIs of
+            Just (mod, filepath) ->
+                concatForM dirs $ \dir -> do
+                    found <- doesFileExist (dir -/- filepath)
+                    return $ if found then [(mod, unifyPath $ dir -/- filepath)]
```

Comment:
`[ (mod, unifyPath $ dir -/- filepath) | found ]` might be less awkward.

Reply:
Fixed.